modules/sdk-coin-sol/src/sol.ts

```diff
@@ -428,6 +459,214 @@ export class Sol extends BaseCoin {
+      isDurableNonceRecovery = true;
+    }
+
+    if (!params.recoveryDestination || !this.isValidAddress(params.recoveryDestination)) {
+      throw new Error('invalid recoveryDestination');
+    }
+
+    // Clean up whitespace from entered values
+    const userKey = params.userKey.replace(/\s/g, '');
+    const backupKey = params.backupKey.replace(/\s/g, '');
+    const bitgoKey = params.bitgoKey.replace(/\s/g, '');
+
+    // Decrypt private keys from KeyCard values
+    let userPrv;
+    try {
+      userPrv = this.bitgo.decrypt({
+        input: userKey,
+        password: params.walletPassphrase,
+      });
+    } catch (e) {
+      throw new Error(`Error decrypting user keychain: ${e.message}`);
+    }
+    const userSigningMaterial = JSON.parse(userPrv) as EDDSAMethodTypes.UserSigningMaterial;
+
+    let backupPrv;
+    try {
+      backupPrv = this.bitgo.decrypt({
+        input: backupKey,
+        password: params.walletPassphrase,
+      });
+    } catch (e) {
+      throw new Error(`Error decrypting backup keychain: ${e.message}`);
+    }
+    const backupSigningMaterial = JSON.parse(backupPrv) as EDDSAMethodTypes.BackupSigningMaterial;
+
+    // Build the transaction
+    const MPC = await EDDSAMethods.getInitializedMpcInstance();
+    const accountId = MPC.deriveUnhardened(bitgoKey, `m/0`).slice(0, 64);
+    const bs58EncodedPublicKey = new SolKeyPair({ pub: accountId }).getAddress();
+
+    const factory = this.getBuilder();
+    const balance = await this.getAccountBalance(bs58EncodedPublicKey);
+    const fee = await this.getFees();
+    const netAmount = balance - (isDurableNonceRecovery ? fee * 2 : fee);
+    let blockhash = await this.getBlockhash();
+    let authority = '';
+
+    if (isDurableNonceRecovery) {
+      const durableNonceInfo = await this.getAccountInfo(params.durableNoncePK);
+      blockhash = durableNonceInfo.blockhash;
+      authority = durableNonceInfo.authority;
+    }
+
+    const txBuilder = factory
+      .getTransferBuilder()
+      .nonce(blockhash)
+      .sender(bs58EncodedPublicKey)
+      .send({ address: params.recoveryDestination, amount: netAmount.toString() })
+      .fee({ amount: fee })
+      .feePayer(bs58EncodedPublicKey);
+    if (isDurableNonceRecovery) {
+      txBuilder.nonce(blockhash, {
+        walletNonceAddress: params.durableNoncePK ? params.durableNoncePK : '',
+        authWalletAddress: authority,
+      });
+    }
+
+    const unsignedTransaction = (await txBuilder.build()) as Transaction;
+
+    // add signature
+    const signatureHex = await EDDSAMethods.getTSSSignature(
+      userSigningMaterial,
+      backupSigningMaterial,
+      'm/0',
+      unsignedTransaction
+    );
+
+    const publicKeyObj = { pub: bs58EncodedPublicKey };
+    txBuilder.addSignature(publicKeyObj as PublicKey, signatureHex);
+
+    if (isDurableNonceRecovery) {
+      // add durable nonce account signature
+      txBuilder.sign({ key: params.durableNonceSK ? params.durableNonceSK : '' });
```

Comment:
Is adding this empty signature key necessary?

Reply:
yeah, i think it is necessary, since we make `durableNonceSK` an optional field, otherwise the linting gives us an error due to the value being possible undefined

Reply:
Would it not be plausible to do `if (isDurableNonceRecovery && params.durableNonceSK)`?
I am curious if the txBuilder will still accept this empty signature as valid.